### ISSUE TITLE
Tests passing on Linux but failing on Windows (yeah weird)

### ISSUE THREAD

**Issue by iShibi** [vitest-dev/vitest#379]
### Describe the bug

I have a project (esm) that uses [`undici`](https://github.com/nodejs/undici) (cjs). Everything works as expected during runtime but for some reason, the tests are failing. By looking at the error, it seems that vitest thinks `module.exports.Something` is undefined when undici does `module.exports.Something = Something`. What's weird is that this only happens on Windows, on Linux the same tests get passed without any issue.

I have created a repro for the issue that I'm facing. Try to run the tests on both Linux and Windows. It would fail on Windows.

### Reproduction

https://github.com/iShibi/issue-repro

### System Info

```shell
System:
  OS: Windows 10 10.0.19043
  CPU: (12) x64 AMD Ryzen 5 4600H with Radeon Graphics
  Memory: 1.67 GB / 7.36 GB
Binaries:
  Node: 16.13.1 - ~\AppData\Local\Volta\tools\image\node\16.13.1\node.EXE
  Yarn: 1.22.11 - ~\AppData\Local\Volta\tools\image\yarn\1.22.11\bin\yarn.CMD
  npm: 8.1.2 - ~\AppData\Local\Volta\tools\image\node\16.13.1\npm.CMD
Browsers:
  Edge: Spartan (44.19041.1266.0), Chromium (96.0.1054.62)
  Internet Explorer: 11.0.19041.1202
npmPackages:
  vitest: ^0.0.124 => 0.0.124
```

### Used Package Manager

npm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [X] Read the [docs](https://vitest.dev/guide).
- [X] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by sheremet-va** [comment]
This is a `mlly` bug: https://github.com/unjs/mlly/pull/28

**Comment by userquin** [comment]
It is working for you `npm run dev`? what version of `node` are you using?

![imagen](https://user-images.githubusercontent.com/6311119/147824037-595d80ea-bf77-4c0d-8b8d-013acc634fcc.png)

You can try with this:

![imagen](https://user-images.githubusercontent.com/6311119/147824204-75f0a4ed-4080-4174-9e49-ae4996c650b1.png)

**Comment by sheremet-va** [comment]
For now you can add this to your `test` config:

```
deps: {
  external: [/undici/],
},
```

**Comment by sheremet-va** [comment]
Fixed in [v0.0.139](https://github.com/vitest-dev/vitest/releases/tag/v0.0.139)
